From b5d8dedca2fbd8b652b762a61c5b92d9db63a2f5 Mon Sep 17 00:00:00 2001
From: Denis Barbier <bouzim@gmail.com>
Date: Thu, 13 Jun 2013 12:33:12 +0200
Subject: [PATCH] Fix loading of TKOpenGl

The TKOpenGl library name was previously returned by the
Graphic3d_GraphicDevice::ShrEnvString method.
We had modified it in the past in order to support debug
libraries on Windows.

Our changes did no more apply cleanly on OCCT 6.6.0, and since
OCCT developers claimed that TKOpenGl is no more loaded at
runtime, our changes had been discarded.

Qb noticed that TKOpenGl is still loaded at runtime, so reinstate
our changes, but this time against Graphic3d::InitGraphicDriver.
---
 src/Graphic3d/Graphic3d.cxx | 8 ++++++++
 1 file changed, 8 insertions(+)

Index: oce/src/Graphic3d/Graphic3d.cxx
===================================================================
--- oce.orig/src/Graphic3d/Graphic3d.cxx
+++ oce/src/Graphic3d/Graphic3d.cxx
@@ -17,6 +17,10 @@
 #include <Aspect_DriverDefinitionError.hxx>
 #include <OSD_Environment.hxx>
 
+#ifdef HAVE_OCE_PATHS_H
+# include <oce-paths.h>
+#endif
+
 //=======================================================================
 //function : InitGraphicDriver
 //purpose  :
@@ -32,6 +36,13 @@ Handle(Graphic3d_GraphicDriver) Graphic3
 
   TCollection_AsciiString aGraphicLibName;
 
+  const char *shr = getenv("CSF_GraphicShr");
+  if (shr != NULL) {
+    aGraphicLibName = shr;
+  } else {
+#ifdef OCE_DEFAULT_CSF_GraphicShr
+  aGraphicLibName = OCE_DEFAULT_CSF_GraphicShr;
+#else
   // Setting the library name. Depends on the platform.
 #if defined(_WIN32) || defined(__WIN32__)
   aGraphicLibName = "TKOpenGl.dll";
@@ -42,6 +53,8 @@ Handle(Graphic3d_GraphicDriver) Graphic3
 #else
   aGraphicLibName = "libTKOpenGl.so";
 #endif
+#endif
+  }
 
   // Loading the library.
   OSD_SharedLibrary aSharedLibrary (aGraphicLibName.ToCString());
